snuba/datasets/processors/functions_processor.py

```diff
@@ -23,7 +23,9 @@ def process_message(
         max_depth_reached = False
 
         profile_id = str(uuid.UUID(message["profile_id"]))
-        timestamp = datetime.utcfromtimestamp(message["timestamp"])
+        now = datetime.utcnow().timestamp()
+        raw_timestamp = max(message.get("timestamp", 0), 0)
+        timestamp = datetime.utcfromtimestamp(raw_timestamp or now)
```

Comment:
I don't think this every applies now, it falls back to 1st January 1970

Reply:
What do you mean? if raw_timestamp is 0 because we received null or a negative value, it will get the `now` value.

Reply:
Sorry, meant to dismiss this. Please ignore.